sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/PhpUnitResultParser.java

```diff
@@ -180,16 +182,22 @@ protected void saveTestReportMeasures(PhpUnitTestReport fileReport, SensorContex
         context.<Integer>newMeasure().on(unitTestFile).withValue(fileReport.getSkipped()).forMetric(CoreMetrics.SKIPPED_TESTS).save();
       }
       double duration = Math.round(fileReport.getTime() * MILLISECONDS);
-      context.<Long>newMeasure().on(unitTestFile).withValue((long)duration).forMetric(CoreMetrics.TEST_EXECUTION_TIME).save();
-      context.<Integer>newMeasure().on(unitTestFile).withValue((int)testsCount).forMetric(CoreMetrics.TESTS).save();
-      context.<Integer>newMeasure().on(unitTestFile).withValue(fileReport.getErrors()).forMetric(CoreMetrics.TEST_ERRORS).save();
-      context.<Integer>newMeasure().on(unitTestFile).withValue(fileReport.getFailures()).forMetric(CoreMetrics.TEST_FAILURES).save();
-      if (testsCount > 0) {
-        double passedTests = testsCount - fileReport.getErrors() - fileReport.getFailures();
-        double percentage = passedTests * PERCENT / testsCount;
-        context.<Double>newMeasure().on(unitTestFile).withValue(ParsingUtils.scaleValue(percentage)).forMetric(CoreMetrics.TEST_SUCCESS_DENSITY).save();
+
+      // fixme SONARPHP-684
+      try {
+        context.<Long>newMeasure().on(unitTestFile).withValue((long) duration).forMetric(CoreMetrics.TEST_EXECUTION_TIME).save();
+        context.<Integer>newMeasure().on(unitTestFile).withValue((int) testsCount).forMetric(CoreMetrics.TESTS).save();
+        context.<Integer>newMeasure().on(unitTestFile).withValue(fileReport.getErrors()).forMetric(CoreMetrics.TEST_ERRORS).save();
+        context.<Integer>newMeasure().on(unitTestFile).withValue(fileReport.getFailures()).forMetric(CoreMetrics.TEST_FAILURES).save();
+        if (testsCount > 0) {
+          double passedTests = testsCount - fileReport.getErrors() - fileReport.getFailures();
+          double percentage = passedTests * PERCENT / testsCount;
+          context.<Double>newMeasure().on(unitTestFile).withValue(ParsingUtils.scaleValue(percentage)).forMetric(CoreMetrics.TEST_SUCCESS_DENSITY).save();
+        }
+        saveTestsDetails(fileReport);
+      } catch (SonarException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "SonarException"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)